react-effects/my-app/src/List.tsx

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { readItems, type Item } from './read';
+
+export function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState<Item[]>([]);
+  const [error, setError] = useState<unknown>();
+
+  useEffect(() => {
+    async function work() {
+      try {
+        const read = await readItems();
+        setItems(read);
+      } catch (error) {
+        setError(error);
+      }
+      setIsLoading(false);
```

Comment:
It would be best to put this in a `finally` clause, though what you have here works, too.